src/types.cpp

```diff
@@ -53,24 +54,36 @@ std::string typestr(Type t)
   }
 }
 
-ProbeType probetype(const std::string &type)
+ProbeType probetype(const std::string &probeName)
 {
-  for (int i = 0; i < sizeof(PROBE_LIST); i++)
-  {
-    if (type == PROBE_LIST[i].name || type == PROBE_LIST[i].abbr)
-      return PROBE_LIST[i].type;
-  }
-  abort();
+  ProbeType retType = ProbeType::invalid;
+
+  auto v = std::find_if(PROBE_LIST.begin(), PROBE_LIST.end(),
+                          [probeName] (const ProbeItem& p) {
```

Comment:
Can we capture probeName by reference here instead
e.g. `[&probeName]`

Probably won't make any sort of difference, but I just prefer to minimise extra copying where we can